src/app/api/utils/featureFlags.ts

```diff
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import appConstants from "../../../appConstants"
+
+type FeatureFlags = Record<
+  "test" |
+  "otherFlag" |
+  "otherFlag2",
+  FeatureFlag | null>;
+
+type FeatureFlag = {
+  name: keyof FeatureFlags;
+  enabled: boolean;
+  description?: string;
+  dependencies?: Array<keyof FeatureFlags>;
+}
+
+let featureFlags: FeatureFlags
+try {
+  featureFlags = JSON.parse(appConstants.FEATURE_FLAGS) as FeatureFlags
+} catch (e) { console.error(e) }
```

Comment:
I think this has a subtle bug here.
Default value of `featureFlags` is `undefined`.
Based on my quick testing, this will throw a `TypeError` if you try accessing a nested property on the `featureFlags` object (like the `isFeatureEnabled()` method below on L25-27.

```js
let featureFlags;
console.log(featureFlags); // undefined
featureFlags['someExperiment']?.enabled
// VM165:1 Uncaught TypeError: Cannot read properties of undefined (reading 'someExperiment')
//    at <anonymous>:1:13
```

I think the workaround is either default `featureFlags` to an empty object `let featureFlags: FeatureFlags = {}` when creating the variable, or setting the fallback empty `featureFlags` object in the `catch` block.

Other workaround might be using one more optional chaining on the Boolean `featureFlags[flag]?.enabled` lookup on L26 below:

```js
return featureFlags?.[flag]?.enabled || false
```